starfish/spots/_detector/local_search_blob_detector.py

```diff
@@ -0,0 +1,423 @@
+        coords = {
+            Features.SPOT_RADIUS: (Features.AXIS, anchor_df[Features.SPOT_RADIUS]),
+            Axes.ZPLANE.value: (Features.AXIS, anchor_df[Axes.ZPLANE]),
+            Axes.Y.value: (Features.AXIS, anchor_df[Axes.Y]),
+            Axes.X.value: (Features.AXIS, anchor_df[Axes.X]),
+            Axes.ROUND.value: (Axes.ROUND.value, rounds),
+            Axes.CH.value: (Axes.CH.value, channels)
+        }
+        intensity_table = IntensityTable(data=data, dims=dims, coords=coords)
+
+        # fill IntensityTable
+        for r in rounds:
+
+            # get intensity data and indices
+            spot_indices = ind[r]
+            intensity_data = round_dataframes[r].loc[spot_indices, 'intensity']
+            channel_index = round_dataframes[r].loc[spot_indices, Axes.CH]
+            round_index = np.full(ind.shape[0], fill_value=r, dtype=int)
+            feature_index = np.arange(ind.shape[0], dtype=int)
+
+            # mask spots that are outside the search radius
+            mask = np.asarray(dist[r] < search_radius)  # indices need not match
+            feature_index = feature_index[mask]
+            channel_index = channel_index[mask]
+            round_index = round_index[mask]
+            intensity_data = intensity_data[mask]
+
+            # need numpy indexing to set values in vectorized manner
+            intensity_table.values[feature_index, channel_index, round_index] = intensity_data
+
+        return intensity_table
+
+    def run(
+        self, data: ImageStack, verbose: bool=False, n_processes: Optional[int]=None
+    ) -> IntensityTable:
+        """Find 1-hot coded spots in data.
+
+        Parameters
+        ----------
+        data : ImageStack
+            Image data containing coded spots.
+        verbose : bool
+            If True, report on progress of spot finding.
+        n_processes : Optional[int]
+            Number of processes to devote to spot finding. If None, will use the number of available
+            cpus (Default None).
+
+        Returns
+        -------
+        IntensityTable
+            Contains detected coded spots.
+
+        """
+        per_tile_spot_results = self._find_spots(data, verbose=verbose, n_processes=n_processes)
+
+        per_round_spot_results = self._merge_spots_by_round(per_tile_spot_results)
+
+        distances, indices = self._match_spots(
+            per_round_spot_results,
+            search_radius=self.search_radius, anchor_round=self.anchor_round
+        )
+
+        # TODO implement consensus seeding (SeqFISH)
+
+        intensity_table = self._build_intensity_table(
+            per_round_spot_results, distances, indices,
+            rounds=data.xarray[Axes.ROUND.value].values, channels=data.xarray[Axes.CH.value].values,
+            search_radius=self.search_radius, anchor_round=self.anchor_round
+        )
+
+        transfer_physical_coords_from_imagestack_to_intensity_table(
```

Comment:
If this gets called through the detect_spots() path you shouldn't need to add this here. detect_spots calls it at the end regardless of the detect method.

Reply:
Unfortunately this does not get called through that pattern; I would need to refactor it but I think I'm more generally going to combine a bunch of the spot detection methods according to #1005